### PR TITLE
!feat: simulate directory in blob list

### PR DESCRIPTION
**Pull request by farnabaz**
resolves nuxt-hub/core#101

This creates a slightly breaking change when delimiter is defined in request

https://github.com/nuxt-hub/core/assets/2047945/ee7627f4-0972-4179-b064-8a76a80ea04a

### CODE REVIEW

**Comment by Atinux** [comment]
Thanks for the PR!

I would like to give a consistent experience for what is returned from the `list()` method and returns:

```ts
{
  blobs: []
  hasMore: boolean
  cursor?: string
  folders?: string[]
}
```

Also, adding the option `folded: true` option instead to return the `folders`:

```ts
// Folded mode
const { blobs, folders } = await hubBlob().list({ folded: true })

// Expanded mode
const { blobs } = await hubBlob().list()
```

By returning `hasMore` and `cursor`, we also fix the pagination issue we have (to be tested):

We should be able to do:

```ts
const { list } = hubBlob()
let hasMore = true
let cursor
 
while (hasMore) {
  const res = await list({
    limit: 10,
    cursor,
  })
  hasMore = res.hasMore
  cursor = res.cursor
}
```

**Comment by farnabaz** [comment]
> Also, adding the option `folded: true` option instead to return the `folders`:

Buckets accepts `delimiter` string to create these folds. I think kipping this delimiter makes more sense, and allows users to customise folding bahavior. WDYT @Atinux ? 
```ts
const { blobs, folders } = await hubBlob().list({ delimiter: true })
```

File: src/runtime/server/utils/blob.ts [hunk 1]
```diff
@@ -149,26 +176,21 @@ export function hubBlob(): HubBlob {
     async list(options: BlobListOptions = { limit: 1000 }) {
```
Comment (Atinux) [inline src/runtime/server/utils/blob.ts:176]:
```suggestion
    async list(options?: BlobListOptions) {
```

File: src/runtime/server/utils/blob.ts [hunk 2]
```diff
@@ -149,26 +176,21 @@ export function hubBlob(): HubBlob {
...
       const resolvedOptions = defu(options, {
         limit: 500,
```
Comment (Atinux) [inline src/runtime/server/utils/blob.ts:178]:
```suggestion
        limit: 1000,
```

File: playground/pages/blob.vue [hunk 3]
```diff
@@ -17,8 +47,12 @@ async function addFile() {
   try {
     const formData = new FormData()
     newFilesValue.value.forEach(file => formData.append('files', file))
+    // TODO: support upload prefix
```
Comment (farnabaz) [inline playground/pages/blob.vue:50]:
```suggestion
```

File: docs/content/docs/2.storage/3.blob.md [hunk 4]
```diff
@@ -238,3 +241,63 @@ interface BlobObject {
...
+    cursor: cursor ? cursor : undefined
+  })
+})
+```
+
+```vue [MyComponent.vue]
+<script setup lang="ts">
+const blobs = ref<BlobObject[]>([])
+const hasMore = ref(true)
+const cursor = ref(undefined)
```
Comment (Atinux) [inline docs/content/docs/2.storage/3.blob.md:276]:
```suggestion
const cursor = ref()
```

File: docs/content/docs/2.storage/3.blob.md [hunk 5]
```diff
@@ -238,3 +241,63 @@ interface BlobObject {
...
+  const { limit, cursor } = await getQuery(event)
+
+  return hubBlob().list({
+    limit: limit ? Number.parseInt(limit) : 10,
+    cursor: cursor ? cursor : undefined
+  })
+})
+```
+
+```vue [MyComponent.vue]
```
Comment (Atinux) [inline docs/content/docs/2.storage/3.blob.md:272]:
```suggestion
```vue [pages/blobs.vue]
```

**Pull request merged**